resdk/resources/collection.py

```diff
@@ -152,11 +143,45 @@ class Collection(BaseCollection):
 
     endpoint = 'collection'
 
+    #: (lazy loaded) list of samples that belong to collection
+    _samples = None
+
     def __init__(self, slug=None, id=None,  # pylint: disable=redefined-builtin
                  model_data=None, resolwe=None):
         """Initialize attributes."""
         BaseCollection.__init__(self, slug, id, model_data, resolwe)
 
+    def update(self):
+        """Clear cache before updating."""
+        self._samples = None
+
+        super(Collection, self).update()
+
+    def add_samples(self, *samples):
+        """Add `samples` objects to the collection."""
+        # Prevent circular import
+        from .sample import get_sample_id
+
+        samples = [get_sample_id(s) for s in samples]
+        for sample in samples:
+            self.resolwe.api.sample(sample).add_to_collection.post({'ids': [self.id]})
```

Comment:
I agree, main problem is that `Collection` is implemented in `resolwe` and `Sample` in `resolwe-bio`. We have to think how to handle such cases in general.

For now I Can add a comment that this has to be improved.